core/src/main/scala/org/apache/spark/scheduler/ShuffleMapTask.scala

```diff
@@ -113,3 +118,24 @@ private[spark] class ShuffleMapTask(
 
   override def toString: String = "ShuffleMapTask(%d, %d)".format(stageId, partitionId)
 }
+
+object ShuffleMapTask extends Logging {
+  private val cache = CacheBuilder.newBuilder()
```

Comment:
Do we need to clear this `cache` at the end of a app ?

Reply:
we don't need to clear, this cache can  recovery itself